### PR TITLE
Fix greytide implants being unintuitive

### LINKED ISSUE #23302: Greytide implants don't drop when cloned nor when de-implanted
#### Description of issue

First one is an oversight, second is technically a feature but a very bullshit one.
(you need to get loyalty implanted to actually drop the role, which makes it impossible for a civilian to "liberate" an implanted dude).

### PR DESCRIPTION
**Pull request by ShiftyRail**
Rather than removing the implant and then adding a loyalty implant, the role is dropped as soon as you remove the implant. This was counter-intuitive, and bugged, as the role was supposed to be dropped as soon as you implanted the person.
You will also lose your greytide implant if you die and get cloned.

On the other hand, adding a loyalty implant on top of the greytide one does nothing.

This second part can be changed if needed.

Closes #23302

[bugfix] [balance]

:cl:
- tweak: You are now free from a greytide implant as soon as the greytide implant is removed, or if you get cloned.

### CODE REVIEW

**Comment by Kurfursten** [comment]
Requiring a loyalty implant at least requires sec to determine what's going on.

This takes us a step deeper into
>just shoot him

**Comment by ShiftyRail** [comment]
Intuitively if I remove the greytide implant from someone I expect them to not be an antag anymore
This is not the case currently

**Comment by nuklearcellphoneg** [comment]
what

what does this pr do

**Comment by ShiftyRail** [comment]
How it works currently : 

```
Greytide implant someone
> he dies and get cloned? Still greytided.
> you remove the implant? Still greytided.
> you loyalty implant them? Still greytided.

>you remove the implant AND loyalty implant them? Bingo
```

How it works after this PR : 
```
>he dies and gets cloned? Un-greytdided.
>you remove the implant? Un-greytided.
```

And loyalty implants have no longer an effect (besides protecting you from being greytided in the first place.)

**Comment by nuklearcellphoneg** [comment]
Okay, please remove loyalty implants preventing greytide conversion, as loyalty to the greytide =/= syndicate, as the greytide both oppose and assist Nanotrasen.

**Comment by ShiftyRail** [comment]
>Okay, please remove loyalty implants preventing greytide conversion, as loyalty to the greytide =/= syndicate, as the greytide both oppose and assist Nanotrasen.

#18108 is the PR that did it. I don't think it is in the scope of the current PR.

The idea is that, from a balance standpoint, it's too trivial to stun/slip a sec off/HoS/Captain and then use the implicit trust these roles have to wreck havoc.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 1]
```diff
@@ -17,6 +17,9 @@
 /obj/item/weapon/implant/proc/activate()
 	return
 
+// What do the implant does when it's removed?
```
Comment (DamianX) [inline code/game/objects/items/weapons/implants/implant.dm]:
```suggestion
// What does the implant do when it's removed?
```

**Comment by Durgen** [comment]
A major debuff to a classic traitor item should at least be paired with a proportional telecrystal discount

**Comment by ShiftyRail** [comment]
>A major debuff to a classic traitor item should at least be paired with a proportional telecrystal discount

Greytide implants already got a buff (2 implants + shades) when loyalty implants were made to block them.

**Comment by juanototo2** [comment]
Lower the cost if you're going to make it into a killnclonesimulator2019 or a tool that can be instantly ignored by a single item that spawns in security roundstart.

**Comment by RealestEstate** [comment]
seconding the TC cost decrease. as it stands it's usually an agent's entire TC count for a box of implants iirc.

**Comment by juanototo2** [comment]
Also, did I mention that after getting killncloned after being confirmed as a greytide implant receiver, you can pop out of the clone pod and immediately say ;X PERSON CONVERTED ME IS TRAITOR GO GET FREE VALIDS GENTLEMEN

**Comment by ShiftyRail** [comment]
I think this might be better in another PR, since the point of this one was to fix the rather unintuitive behaviour of the implants

Do note that you can now have a loyalty implanted greytider (which wasn't possible before).

**Pull request merged**